rules/policy.go

```diff
@@ -83,28 +276,48 @@ func (r *DefaultRuleRenderer) ProtoRuleToIptablesRules(pRule *proto.Rule, ipVers
 				logCxt.Debug("Rule skipped.")
 				return nil
 			}
-
-			markBit, actions := r.CalculateActions(match, &ruleCopy, ipVersion)
+			if usingCIDRBlocks {
+				// The CIDR matches in the rule overflowed and we rendered them
+				// as additional rules, which set the markAllBlocksPass bit on
+				// success.  Add a match on that bit to the calculated rule.
+				match = match.MarkSet(markAllBlocksPass)
+			}
```

Comment:
Not sure this loop works if you have multiple port ranges with `usingCIDRBlocks`.  At the start of the first iteration, the relevant mark bit is markAllBlocksPass, but at the end of the first iteration - and therefore the start of the second, if there is one - the relevant mark bit may be markBit.